Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -511,12 +513,31 @@ WarpX::ApplyInverseVolumeScalingToCurrentDensity (MultiFab* Jx, MultiFab* Jy, Mu
         // these do not include the guard cells.
         std::array<amrex::Real,3> galilean_shift = {0,0,0};
         const std::array<Real, 3>& xyzmin = WarpX::LowerCorner(tilebox, galilean_shift, lev);
+        const Real rmin  = xyzmin[0] + (tbr.type(0) == NODE ? 0. : 0.5*dx[0]);
+        const Real rminr = xyzmin[0] + (tbr.type(0) == NODE ? 0. : 0.5*dx[0]);
+        const Real rmint = xyzmin[0] + (tbt.type(0) == NODE ? 0. : 0.5*dx[0]);
+        const Real rminz = xyzmin[0] + (tbz.type(0) == NODE ? 0. : 0.5*dx[0]);
         const Dim3 lo = lbound(tilebox);
-        const Real rmin = xyzmin[0];
         const int irmin = lo.x;
+        int const ishift_t = (rmint > rmin ? 1 : 0);
+        int const ishift_z = (rminz > rmin ? 1 : 0);
 
         const long nmodes = n_rz_azimuthal_modes;
 
+        // Grow the tileboxes to include the guard cells, except for the
+        // guard cells at negative radius.
+        if (rmin > 0.) {
+           tbr.growLo(0, ngJ);
+           tbt.growLo(0, ngJ);
+           tbz.growLo(0, ngJ);
+        }
+        tbr.growHi(0, ngJ);
+        tbt.growHi(0, ngJ);
+        tbz.growHi(0, ngJ);
+        tbr.grow(1, ngJ);
+        tbt.grow(1, ngJ);
+        tbz.grow(1, ngJ);
+
         // Rescale current in r-z mode since the inverse volume factor was not
         // included in the current deposition.
         amrex::ParallelFor(tbr, tbt, tbz,
```

Comment:
If I understand well, this ParallelFor calls 3 functions that operate on different arrays with different staggering, and something similar is done for rho. Would it be worth it making this a separate function, more general in terms of staggering, so that we could simply call it 4 times? That would clarify the code significantly.

Reply:
That might work, though there are small differences for each component that would have to be handled. Also, this code will be changed further in the near future since there are better methods for handling the scaling near the axis that will be examined. The simplification may be premature.

Reply:
Thanks for the comments!